### PR TITLE
Create javascript-hoisting-scope.md

### PR DESCRIPTION
**Pull request by skatcat31**
## Description

While it sounds like this would be the same as the "What is hoisting in Javascript" it's more a clarification on it and the difference between `let`/`const` and `var`

Adapted from Pluralsight

## What does your PR belong to?

* [x] Questions / Answers
* [ ] Website
* [ ] General / Things regarding the repository (like CI Integration)

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] Enhancement (non-breaking improvement of a question)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

* [x] My code follows the code style of this project.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [x] I have checked that the changes are working properly
* [x] I have checked that there isn't any PR doing the same
* [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: questions/javascript-hoisting-scope.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+### In Javascript, Hoisting happens to Functions and Variable declared with `var`. Where can they be scoped?
```
Comment (flxwu) [inline questions/javascript-hoisting-scope.md]: typo here: *Variables*

File: questions/javascript-hoisting-scope.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+#### Answer
+
+Either Global or Function scoped.
```
Comment (flxwu) [inline questions/javascript-hoisting-scope.md]: Would change that to "either globally or in a function"

File: questions/javascript-hoisting-scope.md (hunk 1 above)
Comment (flxwu) [inline questions/javascript-hoisting-scope.md]: typo here: Variables

File: questions/javascript-hoisting-scope.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+Unlike `let` or `const` they are not Block scoped and can only exist in one of two places because they are assigned a pointed before any evaluation of the code.
```
Comment (flxwu) [inline questions/javascript-hoisting-scope.md]: *block-scoped*

Comment (flxwu) [reply questions/javascript-hoisting-scope.md]: *pointer*

Comment (flxwu) [reply questions/javascript-hoisting-scope.md]: *exist in either on of them*

File: questions/javascript-hoisting-scope.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+#### Good to hear
+
+* There is more than one type of scope in Javascrit(Global, Function, Block)
```
Comment (flxwu) [inline questions/javascript-hoisting-scope.md]: *JavaScript*

File: questions/javascript-hoisting-scope.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+##### Additional links
+
+<!-- Whenever possible, link a more detailed explanation. -->
+
+* [30 Seconds of Interviews: How does Hoisting work in Javascript](https://github.com/fejes713/30-seconds-of-interviews#how-does-hoisting-work-in-javascript)
+
+<!-- tags: (javascript) -->
+
+<!-- expertise: (1) -->
```
Comment (flxwu) [inline questions/javascript-hoisting-scope.md]: This should be an "Easy" question I think

Comment (skatcat31) [reply questions/javascript-hoisting-scope.md]: Is 1 easy or medium? is our score 0 indexed? My bad

Comment (flxwu) [reply questions/javascript-hoisting-scope.md]: yes, 0 indexed

File: questions/javascript-hoisting-scope.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+### In Javascript, Hoisting happens to functions and variables declared with `var`. Where can they be scoped?
+
+#### Answer
+
+Either globally or in a function.
+
+Unlike `let` or `const` they are not block-scoped and can only exist in either one of them because they are assigned a pointer before any evaluation of the code.
```
Comment (flxwu) [inline questions/javascript-hoisting-scope.md]: We could explain this further :smiley:

Comment (skatcat31) [reply questions/javascript-hoisting-scope.md]: probably. Finding the right balance though without repetition is the issue...

**Comment by skatcat31** [comment]
Bringing in comment from gitter:
___

Maybe we should change the question entirely?

### What happens in the following code?

```js
function test(a){
  {
    let b = a;
    var c = b;
  }
  console.log(c);
  console.log(b);
}
```

**Comment by skatcat31** [comment]
I'll be closing this until I decide on a better way to word the question which will be in a new branch

**Pull request closed without merging**